crates/oci-distribution/src/client.rs

```diff
@@ -0,0 +1,394 @@
+//! OCI distribution client
```

Comment:
We should probably add a note that this is not a full implementation yet, but we want it to be